adcm_aio_client/core/objects/cm.py

```diff
@@ -480,15 +486,27 @@ def object(self: Self) -> Object:
     def action(self: Self) -> Action:
         return self._construct(what=Action, from_data=self._data["action"])
 
-    async def wait(self: Self, status_predicate: Callable[[], bool], timeout: int = 30, poll: int = 5) -> None:
-        if self._data["status"] not in (JobStatus.RUNNING, JobStatus.CREATED):
-            return
+    async def wait(
+        self: Self, timeout: int | None, poll_interval: int = 10, exit_condition: Optional[Callable[[], bool]] = None
+    ) -> Self:
+        if self._data["status"] not in ("running", "created"):
+            return self
+
+        if exit_condition is None:
+            exit_condition = self._default_exit_condition
+
+        max_iterations = float("inf") if timeout is None else (timeout // poll_interval)
```

Comment:
Maybe use datetime here? It's more "timeout-friendly", when this implementation reads like "retry" one.

IIRC there's max datetime that's unlikely be reached (analog to `"inf"`, which is good pick).

Also you may define two functions with explicit names like:
- `until_timeout_reached`
- `until_succeed`

One will be bound to `datetime.now()` + `timeout` (when `timeout` is not `None`)

The other one will always return `True` (timeout is `None`)

---

Just a suggestion, won't insist on it